### PR TITLE
82 sections

### LINKED ISSUE #82: Bring Sections into the gem
- [x] Block Text
- [x] FAQ (use The Porch's bootstrap implementation)
- [x] Http Error
- [x] Marquee Text
- [x] Testimonials
- [x] Video
- [x] Video Highlight

This also include any components (Contentful + partials) necessary to accomplish the task.

This card does NOT include the scope to refactor projects that would otherwise use this capability (the places that we're stealing it from). It does include scope verifying that the DOM appropriately renders (style will come later) in the new WM.org repo.

### PR DESCRIPTION
**Pull request by gburgett**
* Adds a number of predefined sections to `wcc-contentful-app` with associated section views
  * `section-block-text`
  * `section-faq`
  * `section-http-error`
  * `section-marquee-text`
  * `section-testimonial`
  * `section-video`
  * `section-video-highlight`
* Adds a rake task to make it easier to import a section definition from a space in the future.

```
$ rake -D
rake app:import_model[model,space]
    Imports a set of models from a space as a single template for the ModelGenerator.
    Usage: rake wcc:import_model[content-type-id,space-id?,extra-content-type-1,extra-content-type-2...]
```

fixes #82

### CODE REVIEW

File: wcc-contentful-app/app/helpers/wcc/contentful/app/section_helper.rb [hunk 1]
```diff
@@ -25,4 +27,41 @@ def section_id(section)
     title = section.try(:bookmark_title) || section.try(:title)
     CGI.escape(title.gsub(/\W+/, '-')) if title.present?
   end
+
+  def markdown(text)
+    options = {
+      filter_html:     true,
+      hard_wrap:       true,
+      link_attributes: { rel: 'nofollow', target: '_blank' },
```
Comment (rorJeremy) [inline wcc-contentful-app/app/helpers/wcc/contentful/app/section_helper.rb]: We're always putting target '_blank' in these links, and I'm sure the issue will come up again where this isn't always the desired behavior. I'm not sure if this should be configurable some how, or if we need to define the behavior based on context, but anyways, I remember this being an issue in the past.

File: wcc-contentful-app/app/views/sections/_testimonials.html.erb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<div class="row">
+  <div class="column">
+    <% section.testimonials&.each do |testimonial|
+      next unless testimonial %>
```
Comment (rorJeremy) [inline wcc-contentful-app/app/views/sections/_testimonials.html.erb]: Not exactly sure why this is necessary? Is it just to handle the scenario where a testimonial could possibly be nil?

File: wcc-contentful-app/lib/generators/wcc/model_generator.rb [hunk 3]
```diff
@@ -5,14 +5,19 @@ class ModelGenerator < Rails::Generators::Base
     source_root File.expand_path('templates', __dir__)
     argument :model, type: :string
 
-    VALID_MODELS = %w[menu page].freeze
+    VALID_MODELS =
+      Dir.glob("#{__dir__}/templates/*")
+        .select { |f| File.directory? f }
+        .map { |f| File.basename f }
+        .sort
+        .freeze
```
Comment (rorJeremy) [inline wcc-contentful-app/lib/generators/wcc/model_generator.rb]: Good idea

File: wcc-contentful-app/spec/generators/wcc/model_generator_spec.rb [hunk 4]
```diff
@@ -7,100 +7,100 @@
 require 'generator_spec'
 require 'timecop'
 
-RSpec.describe Wcc::ModelGenerator, type: :generator do
-  destination Rails.root.join('tmp/generators')
+# RSpec.describe Wcc::ModelGenerator, type: :generator do
```
Comment (rorJeremy) [inline wcc-contentful-app/spec/generators/wcc/model_generator_spec.rb]: What's the plan for this spec? Right now the entire file is commented out.

File: wcc-contentful-app/app/helpers/wcc/contentful/app/section_helper.rb (hunk 1 above)
Comment (gburgett) [reply wcc-contentful-app/app/helpers/wcc/contentful/app/section_helper.rb]: We can override the `markdown` helper in the app, either by creating our own helper that extends from `WCC::Contentful::App::SectionHelper` or by copy-pasting code and tweaking it.

File: wcc-contentful-app/app/views/sections/_testimonials.html.erb (hunk 2 above)
Comment (gburgett) [reply wcc-contentful-app/app/views/sections/_testimonials.html.erb]: Yes - anytime you have a broken link in an array of links, it comes across as nil.  If they unpublish a testimonial without removing it from the testimonial section, we don't want to blow up.

File: wcc-contentful-app/spec/generators/wcc/model_generator_spec.rb (hunk 4 above)
Comment (gburgett) [reply wcc-contentful-app/spec/generators/wcc/model_generator_spec.rb]: whoops, my bad

**Comment by gburgett** [comment]
The folder structure for the generators is to group all things together that logically go with one "concept" - ex. if you install menus, they don't work without also having `dropdownMenu` and `menuButton`.  Similarly, `section-testimonials` doesn't work without also having `testimonials`.

However, it's totally cool to import `section-block-text` into a project without also importing `section-testimonials`.  Maybe we have a smaller site where we don't want all these sections - we just want to pick and choose 2 or 3 of them.  Having them in different folders allows you to `rails g wcc:model` just the sections you want, instead of having to import all sections.

**Pull request merged**